### ISSUE TITLE
Allow referring to versions with duplicate version numbers on the Maven with filter suffixes

### ISSUE THREAD

**Issue by BasiqueEvangelist** [modrinth/code#2644]
### Is your suggested enhancement related to a problem? Please describe.

Right now, it's not possible to cleanly refer to a version if its version number is ambiguous.
[Moonlight Lib](https://modrinth.com/mod/moonlight/versions) is an example - almost all version numbers have two versions associated with them, which means that using a specific version is much clunkier (even Knossos just uses the version ID for some of these versions)

### Describe the solution you'd like

You should be able to suffix a version number with the needed loader/game version name (or multiple!), which will be used to filter the versions with this version number.
Example:
```gradle
// modImplemention "maven.modrinth:moonlight:fwqQVTzC"
modImplemention "maven.modrinth:moonlight:1.19.4-2.4.13-forge"
```

I'm not sure about the specific format of prefix though, as just a simple hyphen could introduce more ambiguity.

### Describe alternatives you've considered

1. Just use version IDs for ambiguous versions
  This is already possible with the existing Maven endpoint. It is kind of unclean though, as it means that you have to comment what the actual version number is in the gradle.properties (and you can't even immediately see the version number if you're looking at the JAR in an IDE)
2. Prefix/Suffix the version ID with the version number, or any tag of your choosing
  Basically what [Curse Maven](https://www.cursemaven.com/) does for project IDs. This does get around the problem of the version number not being visible, but it is a bit clunky, and it necessitates getting the version ID every time you want to upgrade.
3. Put the version ID/filters in the classifier
  This is basically equivalent to my proposal, but the filter is in the classifier instead. My main issue with this proposal is that this makes it harder to attach sources JARs/whatever to versions, as the classifier is now either only the filter, or the real classifier is prefixed/suffixed. This also makes makes putting version-specific data in the POM impossible ~~but to be honest, I don't think anybody used it :sob:~~
